suite/runsuite.cmake

```diff
@@ -546,6 +552,33 @@ if (ARCH_IS_X86 AND UNIX AND (a64_on_x86_only OR NOT arg_automated_ci))
     " OFF ${arg_package} "")
 endif ()
 
+if (ARCH_IS_X86 AND UNIX)
+  # TODO i#3544: Run tests under QEMU
+  set(orig_extra_ctest_args ${extra_ctest_args})
+  set(run_tests OFF)
+  set(prev_optional_cross_compile ${optional_cross_compile})
+  set(prev_run_tests ${run_tests})
+  if (NOT cross_riscv64_linux_only)
+    set(optional_cross_compile ON)
+  endif ()
+  set(ARCH_IS_X86 OFF)
+  # TODO i#3544: Run build tests.
+  set(build_tests "BUILD_TESTS:BOOL=OFF")
+
+  testbuild_ex("riscv64-debug-internal" ON "
+    DEBUG:BOOL=ON
+    INTERNAL:BOOL=ON
+    ${build_tests}
+    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-riscv64.cmake
+    " OFF ${arg_package} "")
+  testbuild_ex("riscv64-release-external" ON "
+    DEBUG:BOOL=OFF
+    INTERNAL:BOOL=OFF
+    ${build_tests}
+    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-riscv64.cmake
+    " OFF ${arg_package} "")
+endif ()
```

Comment:
We perhaps also need to restore the optional_cross_compile and run_tests vars (similar to the lines 480 and 482 above).